GTMAppAuthSwift/Sources/GTMKeychain.swift

```diff
@@ -0,0 +1,362 @@
+      forService: name,
+      accessibility: kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly
+    )
+  }
+
+  /// Retrieves the password `Data` for the given `String` identifier.
+  ///
+  /// - Parameter name: A `String` identifier for the Keychain item.
+  /// - Returns: The password `Data` if found.
+  /// - Throws: An instance of `KeychainWrapper.Error`.
+  func passwordData(forName name: String) throws -> Data {
+    try passwordDataFromKeychain(keychainItemName: name)
+  }
+
+  /// Retrieves the password `Data` for the given `String` identifier.
+  ///
+  /// - Parameters:
+  ///   - name: A `String` identifier for the Keychain item.
+  ///   - usingDataProtectionKeychain: A `Bool` indicating whether to use the data protection
+  ///     keychain on macOS 10.15.
+  /// - Returns: The password `Data` if found.
+  /// - Throws: An instance of `KeychainWrapper.Error`.
+  @available(macOS 10.15, *)
+  func passwordData(forName name: String, usingDataProtectionKeychain: Bool) throws -> Data {
+    try passwordDataFromKeychain(
+      keychainItemName: name,
+      usingDataProtectionKeychain: usingDataProtectionKeychain
+    )
+  }
+
+  private func passwordDataFromKeychain(
+    keychainItemName: String,
+    usingDataProtectionKeychain: Bool = false
+  ) throws -> Data {
+    keychainHelper.useDataProtectionKeychain = usingDataProtectionKeychain
+    return try keychainHelper.passwordData(service: keychainItemName)
+  }
+
+  /// Removes stored password string, such as when the user signs out.
+  ///
+  /// - Parameter name: The Keychain name for the item.
+  /// - Throws: An instance of `KeychainWrapper.Error`.
+  func removePasswordFromKeychain(forName name: String) throws {
+    try removePasswordFromKeychain(keychainItemName: name)
+  }
+
+  /// Removes stored password string, such as when the user signs out. Note that if you choose to
+  /// start using the data protection keychain on macOS, any items previously created will not be
+  /// accessible without migration.
+  ///
+  /// - Parameters:
+  ///   - name: The Keychain name for the item.
+  ///   - usingDataProtectionKeychain: A Boolean value that indicates whether to use the data
+  ///     protection keychain on macOS 10.15+.
+  /// - Throws: An instance of `KeychainWrapper.Error`.
+  @available(macOS 10.15, *)
+  func removePasswordFromKeychain(forName name: String, usingDataProtectionKeychain: Bool) throws {
+    try removePasswordFromKeychain(
+      keychainItemName: name,
+      usingDataProtectionKeychain: usingDataProtectionKeychain
+    )
+  }
+
+  private func removePasswordFromKeychain(
+    keychainItemName: String,
+    usingDataProtectionKeychain: Bool = false
+  ) throws {
+    keychainHelper.useDataProtectionKeychain = usingDataProtectionKeychain
+    try keychainHelper.removePassword(service: keychainItemName)
+  }
+}
+
+// MARK: - Keychain helper
+
+/// A protocol defining the helper API for interacting with the Keychain.
+protocol KeychainHelper {
```

Comment:
Mark as private

Reply:
This cannot be marked private since it's used outside of the file.